### PR TITLE
Move MAX_ERASE_TIME to DT

### PR DESCRIPTION
**Pull request by ikmdani**
These commits remove the huge if block and replace it with a DT API based
retrieval logic. A new element max-erase-time is added which holds the
maximum erase time of a sector or page or half-page of a flash for all the
series of stm32.

### CODE REVIEW

File: dts/arm/st/f0/stm32f0.dtsi [hunk 1]
```diff
@@ -83,6 +83,10 @@
 				label = "FLASH_STM32";
 
 				write-block-size = <2>;
+				/* STM32F0: maximum erase time of 40ms for a
+				 * 2K sector
+				 */
```
Comment (erwango) [inline dts/arm/st/f0/stm32f0.dtsi:88]: A shorter definition would fit in one line.
```suggestion
				/* maximum erase for a 2K sector */
```
Apply everywhere

**Review by erwango — CHANGES_REQUESTED** [review]

File: drivers/flash/flash_stm32.c [hunk 2]
```diff
@@ -431,7 +390,6 @@ static int stm32_flash_init(const struct device *dev)
 			layout[i].pages_size, layout[i].pages_count);
 	}
 #endif
-
```
Comment (erwango) [inline drivers/flash/flash_stm32.c:434]: Keep this line

**Review by erwango — CHANGES_REQUESTED** [review]
Please rearrange commits order so that bindings definition comes before dts edits

**Review by erwango — APPROVED** [review]

**Review by ABOSTM — APPROVED** [review]
LGTM

**Pull request merged**